src/workload.ts

```diff
@@ -135,55 +110,46 @@ export abstract class Workload extends AbstractPod {
     return [...this._matchExpressions];
   }
 
-}
-
-/**
- * A label selector is a label query over a set of resources.
- *
- * @see https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
- */
-export class LabelSelector {
-
   /**
-   * Creates a `matchLabels` entry from the key and value.
-   * Use `applyToTemplate` to also add this label to the pod metadata of the workload.
+   * @internal
    */
-  public static is(key: string, value: string, applyToTemplate?: boolean) {
-    return new LabelSelector(applyToTemplate ?? false, key, [value], undefined);
+  public _toLabelSelector(): k8s.LabelSelector {
+    return { matchExpressions: this._matchExpressions, matchLabels: this._matchLabels };
   }
 
-  /**
-   * Creates a `matchExpressions` "In" entry.
-   */
-  public static in(key: string, values: string[]) {
-    return new LabelSelector(false, key, values, LabelSelectorRequirementOperator.IN);
+  public _toPodSpec(): k8s.PodSpec {
+    const scheduling = this.scheduling._toKube();
+    return {
+      ...super._toPodSpec(),
+      affinity: scheduling.affinity,
+      nodeName: scheduling.nodeName,
+      tolerations: scheduling.tolerations,
+    };
   }
+}
 
-  /**
-   * Creates a `matchExpressions` "NotIn" entry.
-   */
-  public static notIn(key: string, values: string[]) {
-    return new LabelSelector(false, key, values, LabelSelectorRequirementOperator.NOT_IN);
-  }
+/**
+ * Options for `WorkloadScheduling.spread`.
+ */
+export interface WorkloadSchedulingSpreadOptions {
 
   /**
-   * Creates a `matchExpressions` "Exists" entry.
+   * Indicates the spread is optional, with this weight score.
+   *
+   * @default - no weight. spread is assumed to be required.
    */
-  public static exists(key: string) {
-    return new LabelSelector(false, key, undefined, LabelSelectorRequirementOperator.EXISTS);
-  }
+  readonly weight?: number;
+
+}
+
+export class WorkloadScheduling extends pod.PodScheduling {
 
   /**
-   * Creates a `matchExpressions` "DoesNotExist" entry.
+   * Spread the pods in this workload by the topology key.
    */
-  public static doesNotExist(key: string) {
-    return new LabelSelector(false, key, undefined, LabelSelectorRequirementOperator.DOES_NOT_EXIST);
+  public spread(topologyKey: pod.Topology, options: WorkloadSchedulingSpreadOptions = {}) {
+    // a spread is a separation of the pod from itself
```

Comment:
optional: Move this comment into the docstring

Reply:
Moved 👍